query_server/query_operations.py

```diff
@@ -227,76 +227,85 @@ def format_query_response(donors, genomic_query, summary_stats, page, page_size)
     # and then paginate the list of donors, calling donors_with_clinical_data on each before returning
     return fix_dicts(full_data), 200
 
+GENOMIC_TYPE_MAP = {
+    "Variants": "variants",
+    "Transcriptomes (WTS)": "transcriptomes",
+    "Reads (WGS)": "reads",
+    "Genomes": "genomes"
+}
+
+def get_mapped_genomic_types(genomic_data_types):
+    return [GENOMIC_TYPE_MAP.get(dtype) for dtype in genomic_data_types if GENOMIC_TYPE_MAP.get(dtype)]
+    
 @app.route('/query')
-def query(treatment="", primary_site="", drug_name="", systemic_therapy_type="", chrom="", gene="", page=0, page_size=10, assembly="hg38", exclude_programs=[], session_id=""):
-    headers = get_headers()
-
+def query(
+    treatment="", primary_site="", drug_name="", systemic_therapy_type="",
+    chrom="", gene="", page=0, page_size=10, assembly="hg38",
+    exclude_programs=[], genomic_data_types=[], session_id=""
+):
     # NB: We're still doing table joins here, which is probably not where we want to do them
     # We're grabbing (and storing in memory) all the donor data in Katsu with the below request
-
     # Query the appropriate Katsu endpoint
-    url = f"{config.KATSU_URL}/v3/authorized/query/"
 
+    # Note: We get three extra things from /authorized/query that aren't part of the Donors object:
+    # 1) submitter_sample_ids
+    # 2) primary_site
+    # 3) treatment_type
+    # These are used to build the summary information without needing to re-query Katsu
+    # For the purposes of the return value, let's remove all three of these into their own variables
+    headers = get_headers()
+    url = f"{config.KATSU_URL}/v3/authorized/query/"
+ 
+    # Map clinical filters
     param_mapping = [
         (treatment, "treatment_type"),
         (primary_site, "primary_site"),
         (drug_name, "systemic_therapy_drug_name"),
         (systemic_therapy_type, "systemic_therapy_type"),
         (exclude_programs, "exclude_programs")
     ]
-    params = {
-        'page_size': PAGE_SIZE
-    }
-    for param in param_mapping:
-        if param[0] == "" or param[0] == []:
-            continue
-        params[param[1]] = param[0]
+    params = {'page_size': PAGE_SIZE}
+    for val, key in param_mapping:
+        if val not in ("", []):
+            params[key] = val
 
     full_url = f"{url}?{urllib.parse.urlencode(params, doseq=True)}"
     donors_req = requests.get(full_url, headers=headers)
     if not donors_req.ok:
         if donors_req.status_code == 401:
-            # 401 Unauthorized: the token is invalid
-            ret_val = format_query_response([], [], get_summary_stats([], {}, {}), page, page_size)
-            return ret_val[0], 401
+            # 401 Unauthorized: The token is invalid
+            return format_query_response([], [], get_summary_stats([], {}, {}), page, page_size), 401
```

Comment:
```suggestion
            return format_query_response([], [], get_summary_stats([], {}, {}), page, page_size)[0], 401
```
format_query_response returns a 200 normally. The change to this line causes it to return weirdly